src/BMW_E65.cpp

```diff
@@ -152,16 +152,34 @@ void BMW_E65::Task10Ms()
 {
     if(CANWake)
     {
+        uint8_t data[8]; 
+        int rpm = 0;
         if (Ready())
         {
-            uint32_t data[2];
-            int rpm = MAX(750, revCounter) * 4; // rpm value for E65
-
-            data[0] = 0xff595f;
-            data[1] = 0x99800000 | rpm;
-            can->Send(0x0AA, data); //Send on CAN2
+            data[1] = 0x50 | AA1;  //Counter for 0xAA Byte 0
+            data[2] = 0x07;
+            data[6] = 0x94;
+            data[7] = 0x00;
+            rpm =  MAX(750, revCounter) * 4; // rpm value for E65
```

Comment:
```suggestion
            rpm =  MAX(750, revCounter) * 4; // rpm value for E65
            rpm = MIN(rpm, 16383); // prevent an overflow for (very) high RPMs
```

It might make sense to add a check that revCounter does not exceed 16383, otherwise the 16-bit integer in the CAN message will overflow.